File: benchmarks/binary_trees/Trees.java
public class Trees {
    static class TreeNode {
        TreeNode left, right;
        int value;
        TreeNode(int value) { this.value = value; }
        int check() {
            if (left == null) return value;
            return value + left.check() - right.check();
        }
    }

    static TreeNode buildTree(int depth) {
        if (depth == 0) { TreeNode n = new TreeNode(1); return n; }
        TreeNode node = new TreeNode(depth);
        node.left = buildTree(depth - 1);
        node.right = buildTree(depth - 1);
        return node;
    }

    public static void main(String[] args) {
        int maxDepth = 16;
        TreeNode tree = buildTree(maxDepth);
        System.out.println(tree.check());

        int sum = 0;
        int iterations = 1000;
        for (int i = 0; i < iterations; i++) {
            TreeNode t = buildTree(10);
            sum += t.check();
        }
        System.out.println(sum);
    }
}

File: benchmarks/binary_trees/trees.js
class TreeNode {
  constructor(value) {
    this.value = value;
    this.left = null;
    this.right = null;
  }
  check() {
    if (this.left === null) return this.value;
    return this.value + this.left.check() - this.right.check();
  }
}

function buildTree(depth) {
  if (depth === 0) return new TreeNode(1);
  const node = new TreeNode(depth);
  node.left = buildTree(depth - 1);
  node.right = buildTree(depth - 1);
  return node;
}

const maxDepth = 16;
const tree = buildTree(maxDepth);
console.log(tree.check());

let sum = 0;
const iterations = 1000;
for (let i = 0; i < iterations; i++) {
  const t = buildTree(10);
  sum += t.check();
}
console.log(sum);
